networkx/algorithms/bipartite/matching.py

```diff
@@ -141,11 +141,12 @@ def breadth_first_search():
     def depth_first_search(v):
         if v is not None:
             for u in G[v]:
-                if distances[rightmatches[u]] == distances[v] + 1:
-                    if depth_first_search(rightmatches[u]):
-                        rightmatches[u] = v
-                        leftmatches[v] = u
-                        return True
+                if distances[rightmatches[u]] == distances[
+                    v
+                ] + 1 and depth_first_search(rightmatches[u]):
```

Comment:
I assume this was originally written this way for performance reasons, i.e. you only hit the `depth_first_search` call if the first condition (which is much cheaper to compute) is true.

@MridulS this update will preserve the performance correct?

Reply:
I actually don't think this is more readable code even if it still does preserve the performance (which the "and" should do).  The linter `black` makes the resulting if-structure much harder to read than the original double-if structure.  
Can we just focus this PR on removing the loop that creates `unlayered`?